Use v8 to test wasm binaries are valid in test suite binary format checks

kripken opened the pull request.

aheejin commented on scripts/test/shared.py:
```diff
@@ -389,6 +409,16 @@ def binary_format_check(wast, verify_final_result=True, wasm_as_args=['-g'],
+    ]:
+      cmd = [V8] + V8_OPTS + [in_binaryen('scripts', 'validation_shell.js'), '--', 'a.wasm']
+      print '      ', ' '.join(cmd)
+      subprocess.check_call(cmd, stdout=subprocess.PIPE)
```
Maybe print a warning message if v8 does not exist?

aheejin commented on scripts/validation_shell.js:
```diff
@@ -0,0 +1,40 @@
+if (typeof console === 'undefined') {
+  console = { log: print };
+}
+var tempRet0;
```
Where is this used?

aheejin commented on scripts/validation_shell.js:
```diff
@@ -0,0 +1,40 @@
+  if (!binary.buffer) binary = new Uint8Array(binary);
+} else {
+  var args;
+  if (typeof scriptArgs != 'undefined') {
```
What is `scriptArgs`? (I don't know much about the JS part, sorry)

kripken replied:
It's one of the ways various JS shells give us access to the commandline params. I think scriptArgs is for spidermonkey and arguments is for v8 (or maybe the opposite).

aheejin commented on scripts/validation_shell.js:
```diff
@@ -0,0 +1,40 @@
+
+// Test the wasm for validity. The imports aren't there (we can't use a proxy
+// because we don't know the types), but if we get to the imports, that means
+// the wasm is valid.
```
Don't know much about the v8 validation process. Are imports validated after all other things?

kripken replied:
Yes, it compiles first, then uses imports only when creating an instance (at least v8 and sm do - perhaps chakra does lazy validation/compilation?).

Thinking about this again now, I see the code can be much simpler - we can just compile and not even create an instance, so no imports are involved.

kripken commented:
Thanks, that commit should address those comments.

Except for the note about other wasm files in test/ - good point, we should check them too. I'll do it as a followup.

kripken commented:
Turns out it was just the wasm-dis checks - I was worried there were more. So I added those other wasm files in this PR.

Removed one old invalid file while doing so.
